components/event-receiver/event-input-adapters/org.wso2.carbon.event.input.adapter.kafka/src/main/java/org/wso2/carbon/event/input/adapter/kafka/KafkaEventAdapter.java

```diff
@@ -86,25 +87,29 @@ public int hashCode() {
 
     private static ConsumerConfig createConsumerConfig(String zookeeper, String groupId,
                                                        String optionalConfigs) {
-        Properties props = new Properties();
-        props.put(KafkaEventAdapterConstants.ADAPTOR_SUSCRIBER_ZOOKEEPER_CONNECT, zookeeper);
-        props.put(KafkaEventAdapterConstants.ADAPTOR_SUSCRIBER_GROUP_ID, groupId);
-
-        if (optionalConfigs != null) {
-            String[] optionalProperties = optionalConfigs.split(",");
-
-            if (optionalProperties != null && optionalProperties.length > 0) {
-                for (String header : optionalProperties) {
-                    String[] configPropertyWithValue = header.split(":");
-                    if (configPropertyWithValue.length == 2) {
-                        props.put(configPropertyWithValue[0], configPropertyWithValue[1]);
-                    } else {
-                        log.warn("Optional configuration property not defined in the correct format.\nRequired - property_name1:property_value1,property_name2:property_value2\nFound - " + optionalConfigs);
+        try {
+            Properties props = new Properties();
+            props.put(KafkaEventAdapterConstants.ADAPTOR_SUSCRIBER_ZOOKEEPER_CONNECT, zookeeper);
+            props.put(KafkaEventAdapterConstants.ADAPTOR_SUSCRIBER_GROUP_ID, groupId);
+
+            if (optionalConfigs != null) {
+                String[] optionalProperties = optionalConfigs.split(",");
+
+                if (optionalProperties != null && optionalProperties.length > 0) {
```

Comment:
A very minor catch. Here optionalProperties.length > 0 is not required as we are not doing anything particular if optionalProperties.length == 0. Therefore, adding this condition will not do any difference